### PR TITLE
Returning Promise for the identify() to avoid race condition

### PR DESCRIPTION
**Pull request by zihejia**
address https://github.com/mixpanel/mixpanel-react-native/issues/147

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -187,13 +187,24 @@ export class Mixpanel {
      *     Mixpanel using the same disinct_id will be considered associated with the
      *     same visitor/customer for retention and funnel reporting, so be sure that the given
      *     value is globally unique for each individual user you intend to track.
-     *
+     * @returns {Promise} A promise that resolves when the identify is successful.
+     *     It does not return any value.
```
Comment (jaredmixpanel) [inline index.js:191]: Is this a breaking API change?

File: ios/MixpanelReactNative.swift [hunk 2]
```diff
@@ -131,11 +131,12 @@ open class MixpanelReactNative: NSObject {
      
     @objc
     func identify(_ token: String, distinctId: String,
-                  resolver resolve: RCTPromiseResolveBlock,
-                  rejecter reject: RCTPromiseRejectBlock) -> Void {
+                  resolver resolve: @escaping RCTPromiseResolveBlock,
+                  rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
```
Comment (jaredmixpanel) [inline ios/MixpanelReactNative.swift:135]: Do we also need to update the Android side?

Comment (zihejia) [reply ios/MixpanelReactNative.swift:135]: android is a sync call so it should be fine

File: index.js (hunk 1 above)
Comment (zihejia) [reply index.js:191]: it's still compatible with the old way of calling the api.

**Review by jaredmixpanel — APPROVED** [review]
LGTM

**Pull request merged**